src/core/dynamic-user.c

```diff
@@ -641,18 +655,25 @@ int dynamic_user_serialize(Manager *m, FILE *f, FDSet *fds) {
         /* Dump the dynamic user database into the manager serialization, to deal with daemon reloads. */
 
         HASHMAP_FOREACH(d, m->dynamic_users)
-                (void) dynamic_user_serialize_one(d, "dynamic-user", f, fds);
+                (void) dynamic_user_serialize_one(d, "dynamic-user", f, fds, /* index= */ NULL);
 
         return 0;
 }
 
-void dynamic_user_deserialize_one(Manager *m, const char *value, FDSet *fds, DynamicUser **ret) {
+void dynamic_user_deserialize_one(
+                Manager *m,
+                const char *value,
+                FDSet *fds,
+                int *fds_array,
+                size_t n_fds_array,
```

Comment:
why this array? i dont get it? the fdset can store indexes now just fine, so why need a 2nd data structure here?

Reply:
we are getting an array from SCM_RIGHTS

Reply:
sure, but why not turn that immediately into an fdset?

Reply:
Because there's no need to, it can work with the array directly

Reply:
but that means you add all this complexity to every single place we have to deserialize the fd.

I mean, that's what i don't like about the patch: that it changes code *all* *over* the place.

Let's make this clean, and just hide the details in fdset please.

Reply:
ok, changed to use only fdset